lib/client/api.go

```diff
@@ -1424,7 +1424,7 @@ func (tc *TeleportClient) runCommand(
 // runShell starts an interactive SSH session/shell.
 // sessionID : when empty, creates a new shell. otherwise it tries to join the existing session.
 func (tc *TeleportClient) runShell(nodeClient *NodeClient, sessToJoin *session.Session) error {
-	nodeSession, err := newSession(nodeClient, sessToJoin, tc.Env, tc.Stdin, tc.Stdout, tc.Stderr)
+	nodeSession, err := newSession(nodeClient, sessToJoin, tc.Env, tc.Stdin, tc.Stdout, tc.Stderr, tc.useLegacyID(nodeClient))
```

Comment:
isn't server version returned in ping? why have an extra round trip here?

Reply:
if it's not returned, can we start returning it, and if not set, assume that uuid should be old version?

Reply:
basically I want to remove a need for extra roundtrip

Reply:
We discussed this offline, but gonna add my response here for completeness.

The ping endpoint is only called upon login, not upon SSH, which is where this EOF error is occurring. In addition, the ping endpoint returns the version of the proxy, we need the version of the node. Since a connection to the node is already established when the request is made, it should be very fast.